### DISCUSSION TITLE
combining auth-middleware with intl middleware

### DISCUSSION THREAD

**Discussion by mkhavari01** [amannn/next-intl#341] [Q&A]
### Description

while i am trying to protect a route using next-auth and middleware file because i have setup intl in middleware i am confused how to combine the to work correctly 

this is my middleware.ts
```

import createMiddleware from 'next-intl/middleware';

export default createMiddleware({
  // A list of all locales that are supported
  locales: ['fa', 'en'],

  // If this locale is matched, pathnames work without a prefix (e.g. `/about`)
  defaultLocale: 'fa',
  localeDetection: false
});

export const config = {
  // Skip all paths that should not be internationalized
  matcher: ['/((?!api|_next|.*\\..*).*)']
};
```

and i want to combine this middleware which is for next-auth aswell
```
export { default } from "next-auth/middleware";
export const config = {
  matcher: ["/UserPost/:path*"],
};
```

### Mandatory reproduction URL (CodeSandbox or GitHub repository)

https://github.com/vahid-nejad/next-auth-fullstack/blob/main/src/middleware.ts

### Reproduction description

.

### Expected behaviour

.

**Comment by amannn** [comment]
We have docs and a working example of integrating with Auth.js here: https://next-intl-docs.vercel.app/docs/next-13/middleware#example-auth-js.

Hope this helps!

**Reply by oaj** [reply]
It looks like only public pages have i18n

**Comment by MeDeyvid** [comment]
I believe what you are looking for is in this documentation.

https://next-intl-docs.vercel.app/docs/routing/middleware#composing-other-middlewares

**Comment by mkhavari01** [comment]
thank you a lot the doc solved my problem🙏
